Master restart causes eventsd to stop processing

On Salt 2016.3.0 after I restarted the salt-master the events stopped being processed. 
The logs showed the following AssertionError:

```
2016-06-13 08:22:35 [salt.transport.ipc     ][ERROR   ] Exception occurred in Subscriber while handling stream: Already reading
2016-06-13 08:22:35 [salt.log.setup         ][ERROR   ] An un-handled exception was caught by salt's global exception handler:
AssertionError: Already reading
Traceback (most recent call last):
  File "/usr/bin/salt-eventsd", line 118, in <module>
    main()
  File "/usr/bin/salt-eventsd", line 113, in main
    daemon.start()
  File "/usr/lib/pymodules/python2.7/salteventsd/daemon.py", line 301, in start
    super(SaltEventsDaemon, self).start()
  File "/usr/lib/pymodules/python2.7/salteventsd/daemon.py", line 139, in start
    self.run()
  File "/usr/lib/pymodules/python2.7/salteventsd/daemon.py", line 311, in run
    self.listen()
  File "/usr/lib/pymodules/python2.7/salteventsd/daemon.py", line 371, in listen
    ret = event.get_event(full=True)
  File "/usr/lib/python2.7/dist-packages/salt/utils/event.py", line 579, in get_event
    ret = self._get_event(wait, tag, match_func, no_block)
  File "/usr/lib/python2.7/dist-packages/salt/utils/event.py", line 484, in _get_event
    raw = self.subscriber.read_sync(timeout=wait)
  File "/usr/lib/python2.7/dist-packages/salt/transport/ipc.py", line 654, in read_sync
    return ret_future.result()
  File "/usr/lib/python2.7/dist-packages/tornado/concurrent.py", line 215, in result
    raise_exc_info(self._exc_info)
  File "/usr/lib/python2.7/dist-packages/tornado/gen.py", line 230, in wrapper
    yielded = next(result)
  File "/usr/lib/python2.7/dist-packages/salt/transport/ipc.py", line 631, in _read_sync
    raise exc_to_raise  # pylint: disable=E0702
AssertionError: Already reading
```

--- a/salteventsd/daemon.py
+++ b/salteventsd/daemon.py
@@ -374,6 +374,13 @@ class SaltEventsDaemon(Daemon):
             except KeyboardInterrupt:
                 log.info('Received CTRL+C, shutting down')
                 self.stop(signal.SIGTERM, None)
+            except AssertionError:
+                # Incase the master restarts a reconnect needs to happen.
+                event = salt.utils.event.SaltEvent(
+                    self.node,
+                    self.sock_dir,
+                )
+                ret = event.get_event(full=True)
 
             # if we have not received enough events in to reach event_limit
             # and the timer has fired, dump the events collected so far

